### PR TITLE
Charts

### PR DESCRIPTION
**Pull request by Laranto-spb**
Added datepicker
I can not solve 1 rubocop fail for Charts controller (tryed to do separate methods inside index def)
About validation:
1. User can not print date manualy or reset it
2.  Added validation for empty params 
3. Added **flatpickr** datepicker

### CODE REVIEW

File: app/controllers/charts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    set_credits
+    set_debits
+  end
+
+  private
+
+  def get_credit_transactions(categories, start_d, end_d)
+    credit_transaction = []
+    categories.select { |c| c.transaction_type == 'Credit' }.each do |credit|
```
Comment (e-popolamov) [inline app/controllers/charts_controller.rb]: move select to `scope`

File: app/controllers/charts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      transactions = MoneyTransaction.where(person_category_id: credit.person_categories,
+                                            created_at: start_d..end_d).sum(&:amount_value)
+      credit_transaction += [[credit.name, transactions]] if transactions.positive?
+    end
+    credit_transaction
+  end
+
+  def get_debit_transactions(categories, start_d, end_d)
+    debit_transaction = []
+    categories.select { |c| c.transaction_type == 'Debit' }.each do |debit|
```
Comment (e-popolamov) [inline app/controllers/charts_controller.rb]: this move to `scope` too

File: app/helpers/charts_helper.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
```
Comment (e-popolamov) [inline app/helpers/charts_helper.rb]: if this file not be used - please remove it

**Review by e-popolamov — COMMENTED** [review]
Also u have one rubocop issue.
And what be if start_day will set after end_date?

File: app/helpers/charts_helper.rb (hunk 3 above)
Comment (Laranto-spb) [reply app/helpers/charts_helper.rb]: fixed

**Comment by Laranto-spb** [comment]
> Also u have one rubocop issue. And what be if start_day will set after end_date?

Added validation for  start_day after end_date

File: app/controllers/charts_controller.rb (hunk 1 above)
Comment (Laranto-spb) [reply app/controllers/charts_controller.rb]: Done, change to "where"

File: app/controllers/charts_controller.rb (hunk 2 above)
Comment (Laranto-spb) [reply app/controllers/charts_controller.rb]: Done, change to "where"

File: app/controllers/charts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  def set_user_people
+    @people = current_user.people
+  end
+
+  def set_user_categories
+    @categories = current_user.people.collect(&:categories).flatten.uniq
+  end
+
+  def set_credits
```
Comment (e-popolamov) [inline app/controllers/charts_controller.rb]: methods `set_credits` and `set_debits` so needed?

Comment (Laranto-spb) [reply app/controllers/charts_controller.rb]: Fixed

**Review by e-popolamov — APPROVED** [review]

**Comment by e-popolamov** [comment]
Congratulations! You completed all task!

**Pull request merged**